vendor.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+
+	wd, err := os.Getwd()
+	paniconerr(err, "getting working dir")
+
+	dir := ""
+	flag.StringVar(&dir, "dir", wd, "dir that will be recursively walked for deps")
+	flag.Parse()
+
+	gohome := getGoHome()
+	projectdir, err := filepath.Abs(dir)
+	paniconerr(err, fmt.Sprintf("getting absolute path of[%s]", dir))
+
+	if !strings.HasPrefix(projectdir, gohome) {
+		fmt.Println("dir must be inside your GOPATH")
+		os.Exit(1)
+	}
+
+	packages := parseAllDeps(gohome, projectdir)
+	depsGoHome, err := ioutil.TempDir("", "vendor")
+	paniconerr(err, "creating temp dir")
+	defer os.RemoveAll(depsGoHome)
+
+	os.Setenv("GOPATH", depsGoHome)
+	for pkg := range packages {
+		// TODO: could use concurrency here (fan out -> fan in)
+		getPackage(pkg)
+	}
+	os.Setenv("GOPATH", gohome)
+
+	vendorPackages(depsGoHome, projectdir)
+}
+
+func parsePkgDeps(dir string) []string {
+	fileset := token.NewFileSet()
+	pkgsAST, err := parser.ParseDir(fileset, dir, nil, parser.ImportsOnly)
+	paniconerr(err, fmt.Sprintf("parsing dir[%s] for Go file", dir))
+
+	pkgs := []string{}
+
+	for _, pkgAST := range pkgsAST {
+		for _, file := range pkgAST.Files {
+			for _, pkg := range file.Imports {
+				pkgs = append(pkgs, strings.Trim(pkg.Path.Value, "\""))
+			}
+		}
+	}
+
+	return pkgs
+}
+
+func parseProjectDomain(gohome string, rootdir string) string {
+	projectroot := strings.TrimPrefix(rootdir, path.Join(gohome, "src"))
+	return projectroot[1:]
+}
+
+func parseAllDeps(gohome string, rootdir string) map[string]struct{} {
+	deps := map[string]struct{}{}
+	projectRoot := parseProjectDomain(gohome, rootdir)
```

Comment:
In Go semantics, you're getting the project's _import path_ here. The name 'parseProjectDomain' is a bit misleading.

Reply:
I had a really hard time naming this, thanks =)

Reply:
Actually here is where I need the Go "home" (GOPATH or the user local thing) =P. But I will try to think on some other way to do this.